examples/ffi/exporter.cpp

```diff
@@ -1,150 +1,138 @@
-    if (send_result.tag == DDPROF_FFI_SEND_RESULT_FAILURE)
-    {
-        print_error("Failed to send profile: ", send_result.failure);
-        return 1;
-    }
-    else
-    {
-        printf("Response code: %d\n", send_result.http_response.code);
-    }
+int main(int argc, char *argv[]) {
+  if (argc != 2) {
+    printf("Usage: exporter SERVICE_NAME\n");
+    return 1;
+  }
+  const char *api_key = getenv("DD_API_KEY");
+  if (!api_key) {
+    printf("DD_API_KEY environment variable is no set\n");
+    return 1;
+  }
+
+  const auto service = argv[1];
+
+  const ddprof_ffi_ValueType wall_time = {
+      .type_ = to_slice_c_char("wall-time"),
+      .unit = to_slice_c_char("nanoseconds"),
+  };
+
+  const ddprof_ffi_Slice_value_type sample_types = {&wall_time, 1};
+  const ddprof_ffi_Period period = {wall_time, 60};
+  Holder<ddprof_ffi_Profile> profile{
+      ddprof_ffi_Profile_new(sample_types, &period)};
+
+  ddprof_ffi_Line root_line = {
+      .function = {.name = to_slice_c_char("{main}"),
+                   .filename = to_slice_c_char("/srv/example/index.php")},
+      .line = 0,
+  };
+
+  ddprof_ffi_Location root_location = {
+      // yes, a zero-initialized mapping is valid
+      .mapping = {},
+      .lines = {&root_line, 1},
+  };
+
+  int64_t value = 10;
+  const ddprof_ffi_Label label = {
+      .key = to_slice_c_char("language"),
+      .str = to_slice_c_char("php"),
+  };
+  ddprof_ffi_Sample sample = {
+      .locations = {&root_location, 1},
+      .values = {&value, 1},
+      .labels = {&label, 1},
+  };
+  ddprof_ffi_Profile_add(profile, sample);
+
+  Holder<ddprof_ffi_EncodedProfile> encoded_profile{
+      ddprof_ffi_Profile_serialize(profile)};
+
+  ddprof_ffi_EndpointV3 endpoint = ddprof_ffi_EndpointV3_agentless(
+      DDPROF_FFI_CHARSLICE_C("datad0g.com"), to_slice_c_char(api_key));
+  ddprof_ffi_Tag tags[] = {
+      {DDPROF_FFI_CHARSLICE_C("service"), to_slice_c_char(service)},
+  };
+  ddprof_ffi_NewProfileExporterV3Result exporter_new_result =
+      ddprof_ffi_ProfileExporterV3_new(
+          DDPROF_FFI_CHARSLICE_C("native"),
+          ddprof_ffi_Slice_tag{
+              .ptr = tags,
+              .len = sizeof(tags) / sizeof(tags[0]),
+          },
+          endpoint);
+
+  if (exporter_new_result.tag ==
+      DDPROF_FFI_NEW_PROFILE_EXPORTER_V3_RESULT_ERR) {
+    print_error("Failed to create exporter: ", exporter_new_result.err);
+    return 1;
+  }
+
+  Holder<ddprof_ffi_ProfileExporterV3> exporter{exporter_new_result.ok};
+
+  ddprof_ffi_File files_[] = {{
+      .name = DDPROF_FFI_CHARSLICE_C("auto.pprof"),
+      .file = {.ptr = encoded_profile->buffer.ptr,
+               .len = encoded_profile->buffer.len},
+  }};
+
+  ddprof_ffi_Slice_file files = {.ptr = files_,
+                                 .len = sizeof files_ / sizeof *files_};
+
+  ddprof_ffi_Tag additional_tag{}; // Only needed for address
```

Comment:
Having to give an empty object is slightly strange.

Reply:
Could additional_tags be a pointer that way we could just pass null if not needed ?

Reply:
For a rust slice, no, it can't be null. I think have a routine that will return an empty slice in this case -- I'll check and use it here so we can clean that up.

Reply:
This path is correctly checking for a `NULL` across the FFI, so I've updated the example to use `nullptr` (since it is in C++).

Reply:
Thanks !